# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.h
# Commit: 134228f242de
# Full Hash: 134228f242dee041e94824d207c4f0e68b48c83c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-04-01 09:40:15
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Allow high priority tasks to run before timers r=smaug
#   
#   In Gecko, unlike other runnables where the TaskController
#   picks one each at a time, the TimeoutManager allows multiple
#   timers to run sequentially without allowing other runnables
# ==============================================================================

diff -r 41d90165c19a -r 134228f242de dom/workers/WorkerScope.h
--- a/dom/workers/WorkerScope.h	Mon Mar 31 19:49:03 2025 +0000
+++ b/dom/workers/WorkerScope.h	Mon Mar 31 19:49:03 2025 +0000
@@ -387,6 +387,7 @@
   WebTaskScheduler* Scheduler();
   WebTaskScheduler* GetExistingScheduler() const;
   void SetWebTaskSchedulingState(WebTaskSchedulingState* aState) override;
+  bool HasScheduledNormalOrHighPriorityWebTasks() const override;
 
   WebTaskSchedulingState* GetWebTaskSchedulingState() const override {
     return mWebTaskSchedulingState;